be/src/common/daemon.cpp

```diff
@@ -98,10 +106,22 @@ void* tcmalloc_gc_thread(void* dummy) {
         MallocExtension::instance()->GetNumericProperty("generic.current_allocated_bytes", &used_size);
         MallocExtension::instance()->GetNumericProperty("tcmalloc.pageheap_free_bytes", &free_size);
         size_t phy_size = used_size + free_size; // physical memory usage
+        size_t total_bytes_to_gc = 0;
         if (phy_size > config::tc_use_memory_min) {
             size_t max_free_size = phy_size * config::tc_free_memory_rate / 100;
             if (free_size > max_free_size) {
-                MallocExtension::instance()->ReleaseToSystem(free_size - max_free_size);
+                total_bytes_to_gc = free_size - max_free_size;
+            }
+        }
+        size_t bytes_to_gc = gch.bytes_should_gc(MonoTime::Now(), total_bytes_to_gc);
+        if (bytes_to_gc > 0) {
+            size_t bytes = bytes_to_gc;
+            while (bytes >= GCBYTES_ONE_STEP) {
```

Comment:
What if this value is larger? How about 64MB or 128MB？Is there bad impact?

Reply:
just an value i think is not too small or too big, setting it bigger will hold the lock longer.